src/routes/exercise-finished/ExerciseFinishedScreen.tsx

```diff
@@ -59,35 +59,22 @@ type ExerciseFinishedScreenProps = {
 }
 
 const ExerciseFinishedScreen = ({ navigation, route }: ExerciseFinishedScreenProps): ReactElement => {
-  const {
-    exercise,
-    results,
-    disciplineTitle,
-    disciplineId,
-    vocabularyItems,
-    closeExerciseAction,
-    unlockedNextExercise,
-  } = route.params
+  const { exercise, results, disciplineTitle, closeExerciseAction, unlockedNextExercise } = route.params
   const correctResults = results.filter(doc => doc.result === 'correct')
   const percentageOfCorrectResults = correctResults.length / results.length
   const score = calculateScore(results)
 
-  const repeatExercise = (): void =>
-    navigation.navigate(EXERCISES[exercise].screen, {
-      vocabularyItems,
-      disciplineId,
-      disciplineTitle,
-      closeExerciseAction,
-    })
+  const {
+    exercise: notNeededForNavigation1,
+    results: notNeededForNavigation2,
+    unlockedNextExercise: notNeededForNavigation3,
+    ...navigationParams
+  } = route.params
```

Comment:
Because they are not needed for the back navigation to repeat and nextExercise. If we just pass all params to e.g. repeatExercise, we would not pass the correct params to this route